src/avif.c

```diff
@@ -579,14 +579,16 @@ void avifRGBImageSetDefaults(avifRGBImage * rgb, const avifImage * image)
     rgb->maxThreads = 1;
 }
 
-void avifRGBImageAllocatePixels(avifRGBImage * rgb)
+avifResult avifRGBImageAllocatePixels(avifRGBImage * rgb)
 {
     if (rgb->pixels) {
         avifFree(rgb->pixels);
     }
 
     rgb->rowBytes = rgb->width * avifRGBImagePixelSize(rgb);
     rgb->pixels = avifAlloc((size_t)rgb->rowBytes * rgb->height);
+    AVIF_CHECKERR(rgb->pixels, AVIF_RESULT_OUT_OF_MEMORY);
```

Comment:
Nit: Set `rgb->rowBytes` to 0 before returning `AVIF_RESULT_OUT_OF_MEMORY`.

A simple way to do this is to move line 588 after this line. But I think we should also set `rgb->rowBytes` to 0 at line 585.

Reply:
I used `avifRGBImageFreePixels(rgb)` directly, it is simpler.